modules/coach/src/main/CoachPager.scala

```diff
@@ -18,17 +23,64 @@ final class CoachPager(
   import CoachPager._
   import BsonHandlers._
 
-  def apply(lang: Option[Lang], order: Order, page: Int): Fu[Paginator[Coach.WithUser]] = {
-    val adapter = new Adapter[Coach](
-      collection = coll,
-      selector = listableSelector ++ lang.?? { l =>
-        $doc("languages" -> l.code)
-      },
-      projection = none,
-      sort = order.predicate
-    ) mapFutureList withUsers
+  def apply(
+      lang: Option[Lang],
+      order: Order,
+      country: Option[Country],
+      page: Int
+  ): Fu[Paginator[Coach.WithUser]] = {
+    def selector = listableSelector ++ lang.?? { l => $doc("languages" -> l.code) }
+
+    val adapter = country match {
+      case Some(_) => new AdapterLike[Coach.WithUser] {
+        def nbResults: Fu[Int] = coll.secondaryPreferred.countSel(selector)
+
+        def slice(offset: Int, length: Int): Fu[List[Coach.WithUser]] =
+          coll
+            .aggregateList(length, readPreference = ReadPreference.secondaryPreferred) { framework =>
+              import framework._
+              Match(selector) -> List(
+                Sort(
+                  order match {
+                    case Alphabetical  => Ascending("_id")
+                    case NbReview      => Descending("nbReview")
+                    case LichessRating => Descending("user.rating")
+                    case Login         => Descending("user.seenAt")
+                  }
+                ),
+                PipelineOperator(
+                  $doc(
+                    "$lookup" -> $doc(
+                      "from"         -> userRepo.coll.name,
+                      "localField"   -> "_id",
+                      "foreignField" -> "_id",
+                      "as"           -> "coach"
```

Comment:
should be `"as" -> "user"` since it's a user document we're getting

Reply:
I've set it to `"as" -> "_user"` because I think it was conflicting with something when it was set to `"user"` (it wasn't showing any users when I used the country selection)